File: JAVA/BinarySearch.java

public class BinarySearch {


    public static int binarySearch(int array[], int value) {

        //declare the left index
        int Left = 0;
        int n = array.length;
        //declare the right index
        int Right = n - 1;
        int result = 0;
        for (int i = 0; i < n; i++) {
            if (Left > Right)
                result = -1;
            else {
                //set the middle index of the array
                int mid = (Left + Right) / 2;
                
                if (array[mid] < value)
                    Left = mid + 1;
                if (array[mid] > value)
                    Right = mid - 1;
                if (array[mid] == value)
                    result = mid;
            }
        }
        return result;
    }

    public static void main(String[] args) {
        //sorted array of items
        int[] arr = { 2, 5, 8, 12, 16, 23, 38, 56, 72, 91 };
        //value we need to search
        int value = 23;

        //call the binary search method
        int result = binarySearch(arr, value);
        if (result == -1) {
            System.out.println(value + " is not found");
        } else {
            System.out.println(value + " is found in index " + result);
        }

    }
}

File: C++ Programming/Binarysearch.cpp
#include <iostream>
using namespace std;

int binarySearch(int arr[], int n, int x) 
{
    int left = 0, right = n - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == x)
            return mid;

        if (arr[mid] < x)
            left = mid + 1;
        else
            right = mid - 1;
    }

    return -1;
}

int main() 
{
    int arr[] = {2, 3, 4, 10, 40};
    int n = sizeof(arr) / sizeof(arr[0]);
    int x = 10;
    int result = binarySearch(arr, n, x);
    
    if(result != -1)
        cout << "Element found at index " << result;
    else
        cout << "Element not found in array";
    
    return 0;
}
